### PR TITLE
Issue211 diagram types

### PR DESCRIPTION
**Pull request by philip-alldredge**
Solves issue #211

### CODE REVIEW

**Comment by philip-alldredge** [comment]
@RyanMcilnay  This fixes #211. The documentation has been updated. Please review that as well.

File: org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java [hunk 1]
```diff
@@ -50,13 +60,25 @@
...
-		final DiagramConfigurationBuilder configBuilder = new DiagramConfigurationBuilder(
-				ageDiagram.getConfiguration());
+
+		// Set the diagram type
+		final DiagramType diagramType;
+		final String diagramTypeId = mmDiagram.getConfig() == null || mmDiagram.getConfig().getType() == null
+				? CustomDiagramType.ID
+						: mmDiagram.getConfig().getType();
+		diagramType = extRegistry.getDiagramTypeById(diagramTypeId)
+				.orElseGet(() -> new UnrecognizedDiagramType(diagramTypeId));
```
Comment (RyanMcilnay) [inline org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java]: Is there a reason diagramType is created above where it is initialized?

Comment (philip-alldredge) [reply org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java]: No, that is likely just an artifact of some of the revisions while writing the code. Just pushed a fix.

File: org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java [hunk 2]
```diff
@@ -126,16 +149,19 @@ private static long getMaxId(final org.osate.ge.diagram.DiagramElement de) {
...
-			createElement(m, container, mmContainer, mmElement, usedIdSet);
+			createElement(m, contentFilterProvider, container, mmContainer, mmElement, usedIdSet,
+					usingLegacyContentFilters);
 		}
 	}
 
-	private static void createElement(final DiagramModification m, final DiagramNode container,
-			final org.osate.ge.diagram.DiagramNode mmContainer, final org.osate.ge.diagram.DiagramElement mmChild,
-			final Set<Long> usedIdSet) {
+	private static void createElement(final DiagramModification m, final ContentFilterProvider contentFilterProvider,
```
Comment (RyanMcilnay) [inline org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java]: mmContainer does not seem to be used in createElement.

Comment (philip-alldredge) [reply org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java]: Pushed a fix. Not sure how long that Good catch.

File: org.osate.ge/src/org/osate/ge/internal/diagram/runtime/boTree/DefaultTreeUpdater.java [hunk 3]
```diff
@@ -146,48 +114,57 @@ public BusinessObjectNode expandTree(final DiagramConfiguration configuration, f
...
+				if (contextBo == null) {
 					final String contextLabel = refService.getLabel(configuration.getContextBoReference());
-					throw new RuntimeException(
-							"Unable to find context business object: "
-									+ (contextLabel == null ? configuration.getContextBoReference() : contextLabel));
+					throw new RuntimeException("Unable to find context business object: "
+							+ (contextLabel == null ? configuration.getContextBoReference() : contextLabel));
 				}
 
-				potentialBusinessObjects = Collections.singleton(contextBo);
```
Comment (RyanMcilnay) [inline org.osate.ge/src/org/osate/ge/internal/diagram/runtime/boTree/DefaultTreeUpdater.java]: Leftover references to potentialBuisnessObjects commented out.

File: org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToImplementationDiagramHandler.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+		final DiagramElement selectedElement = selectedDiagramElements.get(0);
+		final Object bo = selectedElement.getBusinessObject();
+
+		final DiagramService diagramService = Objects.requireNonNull(Adapters.adapt(activeEditor, DiagramService.class),
+				"Unable to retrieve diagram service");
+
+		final ComponentImplementation ci = Objects.requireNonNull(getComponentImplementation(bo),
+				"Unable to retrieve component implementation");
+		;
```
Comment (RyanMcilnay) [inline org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToImplementationDiagramHandler.java]: Extra semicolon

**Comment by RyanMcilnay** [comment]
Done reviewing

File: org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToImplementationDiagramHandler.java (hunk 4 above)
Comment (philip-alldredge) [reply org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToImplementationDiagramHandler.java]: Pushed fix.

**Comment by philip-alldredge** [comment]
@RyanMcilnay the comments should be addressed.

**Comment by RyanMcilnay** [comment]
@philip-alldredge Got it

**Pull request merged**